ci/check-protobuf-stability.sh

```diff
@@ -89,26 +89,34 @@ USAGE
   exit
   ;;
 --stable)
-  if [[ $TARGET == "main" ]]; then
-    # Check against the most recent stable tag.
-    # This check runs only on main or PRs targeting main.
-    #
-    # This check does not need to run on release branch commits because
-    # they are built sequentially, so no conflicts are possible and the per-PR
-    # check is enough.
-    readonly LATEST_STABLE_TAG="$(git tag | grep -v "snapshot" | sort -V | tail -1)"
-    # The v1.17 stable release includes the buf config file with the default name `buf.yml`.
-    # Starting with v1.18 we have multiple buf config files.
-    if [[ $LATEST_STABLE_TAG =~ "v1.17."* ]]; then
-      BUF_CONFIG_UPDATED=false
-    else
-      BUF_CONFIG_UPDATED=true
-    fi
-    BUF_GIT_TARGET_TO_CHECK=".git#tag=${LATEST_STABLE_TAG}"
-    check_protos
+  # Check against the highest stable tag according to semver.
+  #
+  # This check does not need to run on release branch commits because
+  # they are built sequentially, so no conflicts are possible and the per-PR
+  # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
+  LATEST_STABLE_TAG=""
+  if [[ "${TARGET}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
+    readonly BRANCH_SUFFIX="${TARGET#release/}"
+    readonly MINOR_VERSION="${BRANCH_SUFFIX%.x}"
+    readonly NEXT_MINOR_VERSION=$(semver bump minor "$MINOR_VERSION.0")
+    readonly STABLE_TAGS=($(git tag | grep "v.*" | grep -v "snapshot" | sort -V))
+    LATEST_STABLE_TAG="$(for TAG in "${STABLE_TAGS[@]}"; do if [[ $(semver compare "${TAG#v}" "$NEXT_MINOR_VERSION") == "-1" ]]; then echo "$TAG"; fi; done | tail -1)"
```

Comment:
```suggestion
    LATEST_STABLE_TAG_FOR_TARGET="$(
      for STABLE_TAG in "${ALL_STABLE_TAGS[@]}"; do
        if [[ $(semver compare "${STABLE_TAG#v}" "$TARGET_BRANCH_MINOR_VERSION_PLUS_ONE") == "-1" ]]; then
          echo "$STABLE_TAG";
        fi;
      done | tail -1)"
```

Reply:
As discussed offline, I did let myself omit the renamings to keep them short, as well as I find them a bit redundant, as they keep repeating `TARGET_BRANCH` already in the scope of using `TARGET_BRANCH/RELEASE`. @fabiotudone-da, it would be great to get a stamp from you :)